### PR TITLE
Add clojure.core/fn to resolve-macro-as options and sort options

### LINKED ISSUE #1094: Add clojure.core/fn to resolve-macro-as options and sort options

### PR DESCRIPTION
**Pull request by bpringe**
Closes #1094 

- [x] I created an issue to discuss the problem I am trying to solve or an open issue already exists.
- [x] I added a new entry to [CHANGELOG.md](https://github.com/clojure-lsp/clojure-lsp/blob/master/CHANGELOG.md)
- ~[ ] I updated documentation if applicable (`docs` folder)~

### CODE REVIEW

**Comment by bpringe** [comment]
@ericdallo I don't know if you wanted to leave the ordering that existed. If so I can remove the sorting, but also, where should I put the `fn` option?

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 ## Unreleased
 
 - Editor
- - Restored #995 Improve element selected on `textDocument/hover` (previously reverted) with a fix that keeps it working for Calva even after a syntax error is introduced.
+  - Restored #995 Improve element selected on `textDocument/hover` (previously reverted) with a fix that keeps it working for Calva even after a syntax error is introduced.
+  - Add clojure.core/fn to resolve-macro-as options and sort options
```
Comment (ericdallo) [inline CHANGELOG.md:7]: Could you link #1094 in the end of the sentence?

**Review by ericdallo — COMMENTED** [review]
@bpringe the sorting was intended to be what people most lint-as, so using `sort` here don't make much sense to me, I think fn can go after `let` and before `for`, WDYT?

File: CHANGELOG.md (hunk 1 above)
Comment (bpringe) [reply CHANGELOG.md]: Done!

**Comment by bpringe** [comment]
That makes sense. I reverted the sort.

File: lib/src/clojure_lsp/feature/resolve_macro.clj [hunk 2]
```diff
@@ -57,7 +57,8 @@
              (update-macro-resolve-for-config (symbol resolved-full-symbol-str) full-symbol))))))
 
 (def ^:private known-full-symbol-resolve
-  ["clojure.core/def"
+  ["clojure.core/fn"
```
Comment (ericdallo) [inline lib/src/clojure_lsp/feature/resolve_macro.clj:60]: I think lint as fn is less common than lint as def or defn or let, could you move it to below `clojure.core/let`?

Comment (bpringe) [reply lib/src/clojure_lsp/feature/resolve_macro.clj]: Oh, sorry, I forgot about that earlier! Fixed.

**Comment by ericdallo** [comment]
@bpringe sorry, could you fix conflicts so we can merge this?

**Comment by bpringe** [comment]
@ericdallo Done! Sorry for the delay.

**Comment by ericdallo** [comment]
Thank you!

**Pull request merged**